unstable/wlr-layer-shell-unstable-v1.xml

```diff
@@ -203,21 +204,85 @@
       <arg name="left" type="int"/>
     </request>
 
+    <enum name="keyboard_interactivity" since="4">
+      <description summary="types of keyboard interaction possible for a layer shell surface">
+        Types of keyboard interaction possible for layer shell surfaces. The
+        rationale for this is twofold: (1) some applications are not interested
+        in keyboard events and not allowing them to be focused can improve the
+        desktop experience; (2) some applications will want to take exclusive
+        keyboard focus.
+      </description>
+
+      <entry name="no_focus" value="0">
+        <description summary="no keyboard focus is possible">
+          This value indicates that this surface is not interested in keyboard
+          events and the compositor should never assign it the keyboard focus.
+
+          This is the default value, set for newly created layer shell surfaces.
+
+          This is useful for e.g. desktop widgets that display information or
+          only have interaction with non-keyboard input devices.
+        </description>
+      </entry>
+      <entry name="exclusive" value="1">
+        <description summary="request exclusive keyboard focus">
+          Request exclusive keyboard focus if this surface is above the shell surface layer.
+
+          For layers above the shell surface layers, the seat will always give
+          exclusive keyboard focus to the top-most layer which has keyboard
+          interactivity set to exclusive. If this layer contains multiple
+          surfaces with keyboard interactivity set to exclusive, the compositor
+          determines the one receiving keyboard events in an implementation-
+          defined manner. In this case, no guarantee is made when this surface
+          will receive keyboard focus (if ever).
+
+          For layers below the shell surface layer, the seat will use
+          normal focus semantics.
```

Comment:
It doesn't seem like this is backwards-compatible. I think e.g. Sway never gives focus in this case.

Reply:
Then sway isn't compliant with the current version of the protocol ;)
https://github.com/swaywm/wlr-protocols/blob/master/unstable/wlr-layer-shell-unstable-v1.xml#L209

Reply:
Indeed, you're correct. This makes the change in this PR look less scary, because it just allows top/overlay layers to get the same behaviour as bottom/background layers which have keyboard interactivity set to 1.

I'd still prefer to allow compositors to not give focus if they don't want to, though.

Reply:
I'm changing this to optional. With this change, clients that want normal focus can just use the next option.